core/chaincode/lifecycle/deployedcc_infoprovider_test.go

```diff
@@ -236,6 +240,114 @@ var _ = Describe("ValidatorCommitter", func() {
+
+		Context("when state range scan returns an error", func() {
+			BeforeEach(func() {
+				fakeQueryExecutor.GetStateRangeScanIteratorReturns(nil, fmt.Errorf("rangescan-error"))
+			})
+
+			It("returns the error", func() {
+				_, err := vc.AllChaincodesInfo("channel-name", fakeQueryExecutor)
+				Expect(err).To(MatchError("could not query namespace metadata: could not get state range for namespace namespaces: could not get state iterator: rangescan-error"))
+			})
+		})
+
+		Context("when get state returns an error", func() {
+			BeforeEach(func() {
+				fakeQueryExecutor.GetStateReturns(nil, fmt.Errorf("getstate-error"))
+			})
+
+			It("returns the error", func() {
+				_, err := vc.AllChaincodesInfo("channel-name", fakeQueryExecutor)
+				Expect(err).To(MatchError("could not get info about chaincode: could not deserialize metadata for chaincode cc-name: could not query metadata for namespace namespaces/cc-name: getstate-error"))
+			})
+		})
+
+		Context("when LegacyProvider.AllChaincodesInfo returns DeployedChaincodeInfo", func() {
+			BeforeEach(func() {
+				fakeLegacyProvider.AllChaincodesInfoReturns([]*ledger.DeployedChaincodeInfo{{
+					Name:     "legacy-ccname",
+					Hash:     []byte("hash"),
+					Version:  "cc-version",
+					IsLegacy: true,
+				}}, nil)
+				fakeQueryExecutor.GetStateRangeScanIteratorReturns(&mock.ResultsIterator{}, nil)
+			})
+
+			It("returns deployed chaincodes info for legacy chaincodes", func() {
+				res, err := vc.AllChaincodesInfo("testchannel", fakeQueryExecutor)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(res).To(HaveLen(1))
+				Expect(res[0]).To(Equal(&ledger.DeployedChaincodeInfo{
+					Name:     "legacy-ccname",
+					Hash:     []byte("hash"),
+					Version:  "cc-version",
+					IsLegacy: true,
+				}))
+				Expect(fakeLegacyProvider.AllChaincodesInfoCallCount()).To(Equal(1))
+				channelID, qe := fakeLegacyProvider.AllChaincodesInfoArgsForCall(0)
+				Expect(channelID).To(Equal("testchannel"))
+				Expect(qe).To(Equal(fakeQueryExecutor))
+			})
+		})
+
+		Context("when LegacyProvider.AllChaincodesInfo returns an error", func() {
+			BeforeEach(func() {
+				fakeLegacyProvider.AllChaincodesInfoReturns(nil, fmt.Errorf("chaincode-info-error"))
+			})
+
+			It("passes through to the legacy impl", func() {
+				_, err := vc.AllChaincodesInfo("testchannel", fakeQueryExecutor)
+				Expect(err).To(MatchError("chaincode-info-error"))
+			})
+		})
+
+		Context("when the data is corrupt", func() {
+			BeforeEach(func() {
+				fakePublicState["namespaces/fields/cc-name/ValidationInfo"] = []byte("garbage")
+			})
+
+			It("wraps and returns that error", func() {
+				_, err := vc.AllChaincodesInfo("channel-name", fakeQueryExecutor)
+				Expect(err).To(MatchError("could not get info about chaincode: could not deserialize chaincode definition for chaincode cc-name: could not unmarshal state for key namespaces/fields/cc-name/ValidationInfo: proto: can't skip unknown wire type 7"))
+			})
+		})
+	})
+
```

Comment:
As discussed in the [comment](https://github.com/hyperledger/fabric/pull/1331/files#r430630953), better to add a test where there is an overlap in the results from `_lifecycle` and `lscc`